packages/dds/map/src/directory.ts

```diff
@@ -697,10 +700,14 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         this.messageHandlers.set(
             "clear",
             {
-                process: (op: IDirectoryClearOperation, local, localOpMetadata) => {
+                process: (msg: ISequencedDocumentMessage, op: IDirectoryClearOperation, local, localOpMetadata) => {
                     const subdir = this.getWorkingDirectory(op.path) as SubDirectory | undefined;
                     if (subdir) {
-                        subdir.processClearMessage(op, local, localOpMetadata);
+                        if(subdir.clientIds.includes(msg.clientId)
```

Comment:
track a lists of client ids, as multiple clients could create the same subdir, and they should all be able to see it if they created it

Reply:
we probably need to do something here for the create->delete->create case to get consistent interleaving's